src/codemagic/tools/_app_store_connect/action_groups/builds_action_group.py

```diff
@@ -39,3 +52,137 @@ def get_build_pre_release_version(self, build_id: ResourceId, should_print: bool
             self.api_client.builds.read_pre_release_version,
             should_print,
         )
+
+    @cli.action(
+        'add-beta-test-info',
+        BuildArgument.BUILD_ID_RESOURCE_ID,
+        BuildArgument.BETA_BUILD_LOCALIZATIONS,
+        BuildArgument.LOCALE_DEFAULT,
+        BuildArgument.WHATS_NEW,
+        action_group=AppStoreConnectActionGroup.BUILDS)
+    def add_beta_test_info(self,
+                           build_id: ResourceId,
+                           beta_build_localizations: Optional[Types.BetaBuildLocalizations] = None,
+                           locale: Optional[Locale] = None,
+                           whats_new: Optional[Types.WhatsNewArgument] = None):
+        """
+        Add localized What's new (what to test) information
+        """
+
+        beta_test_info_items = [*beta_build_localizations.value] if beta_build_localizations else []
+
+        if whats_new:
+            beta_test_info_items.append(BetaBuildInfo(whats_new=whats_new.value, locale=locale))
+
+        for item in beta_test_info_items:
+            self.create_beta_build_localization(build_id=build_id, locale=item.locale, whats_new=item.whats_new)
+
+    @cli.action(
+        'submit-to-testflight',
+        BuildArgument.BUILD_ID_RESOURCE_ID,
+        action_group=AppStoreConnectActionGroup.BUILDS)
+    def submit_to_testflight(
+            self, build_id: ResourceId, max_build_processing_wait: Optional[Types.MaxBuildProcessingWait] = None):
+        """
+        Submit build to TestFlight
+        """
+
+        # Workaround to support overriding default value by environment variable.
+        if max_build_processing_wait:
+            max_processing_minutes = max_build_processing_wait.value
+        else:
+            max_processing_minutes = Types.MaxBuildProcessingWait.default_value
+
+        app = self.api_client.builds.read_app(build_id)
```

Comment:
Actually both app and build resource can be acquired using a single query when you use the `include` parameter. However, currently there is no implementation for using this. But it would be fairly simple to add - just modify `Builds` resource manager with 
```python
    def read_with_include(
            self,
            build: Union[LinkedResourceData, ResourceId],
            include_type: Type[IncludedResource],
    ) -> Tuple[Build, IncludedResource]:
        """
        https://developer.apple.com/documentation/appstoreconnectapi/read_build_information
        """
        included_field = self._get_include_field_name(include_type)
        build_id = self._get_resource_id(build)
        response = self.client.session.get(
            f'{self.client.API_URL}/builds/{build_id}',
            params={'include': included_field},
        ).json()
        return Build(response['data']), include_type(response['included'][0])
```
and make it to be in line with `PreReleaseVersions` implementation for included resources.